lib/plugins/EPrints/Plugin/Export/OPENAIRE.pm

```diff
@@ -203,7 +203,41 @@ sub xml_dataobj
 			$topcontent->appendChild( $creator);
 		  }
 	}
-	$response->appendChild( $topcontent );	
+
+        #EDITORS
+        if ($dataobj->exists_and_set("editors") && !$dataobj->exists_and_set("creators") && !$dataobj->exists_and_set("corp_creators")){
```

Comment:
why would we limit the export of editors to those records that don't have creators or corp_creators?  It would be better to include it regardless, for a more complete and comprehensive export.

Reply:
This was mainly to ensure that edited books with no authors turned up something in creators so just trying to use a narrow scope for accuracy but I definitely have no objections to removing those limiting parts of the if statement!

Reply:
I see, so I think I misunderstood the context initially.  According to the guidelines for literature repositories, version 4, the editor should go into the contributor field, with the role specified as "editor".  see: https://openaire-guidelines-for-literature-repository-managers.readthedocs.io/en/v4.0.0/field_contributor.html  Editor is listed as an example.  What you are trying to do is to place the editor as a creator, when nothing else is present.  That's a really specific context.  I think what we need is, if the editor is present, by default, it should be listed as a contributor, not as a creator.  That leaves the question of what to do when there are no creators listed, just an editor.  That's an unusual use case.  Still, it seems incorrect to place the editor into the wrong field to resolve the issue of no creators.  What exactly is the issue that we would be solving there?  Can you please explain a little more?  I realize that creators are mandatory, but if there are not there in the metadata, then you haven't "resolved" anything by placing a contributor name there instead.  I think that we need to export the editors as contributors.

Reply:
Editors are not listed as the "primary" entry (i.e., 'creator') in RDA or DC, so I would not recommend that we take that route.  It seems like the issue is that of no "creators" and only "contributors" in some records, clashing slightly with the mandatory creator in OpenAIRE guidelines.  However, I believe it's the OpenAIRE guidelines that need to be flexible here. .  There are other types of records for which no "creator" is present, and only "contributors", for example, in the context of film and audio recordings.  Who is the "creator" of a feature film?  Often, these types of creative works have many contributors, and it's not recommended to pick one (for example, the film's director or producer) as the creator.  Instead, there are works that have contributors only, and no creators.  It's the OpenAIRE guidelines that need to deal with that reality.

Reply:
I'm going to go ahead and prepare a commit for editors to be exported as contributors.

Reply:
I agree that by and large editors should be in the contributors field. 
I think in the very specific case of edited books, editors can reasonably be considered to be the creators of the output, though I accept that's a question of semantics! My opinion would be that the best approach is adding a further if clause and only putting editors into the creators field if the output is that item type.

Reply:
According to RDA (Resource Description and Access) standard for descriptive cataloging, just like in AACR2, even in the case of edited books, the editors are not to have a primary access entry/ creator role and instead a secondary / contributor.  The similar role that would get a creator according to RDA is that of a compiler, but that is defined as "An agent responsible for creating a new work such as a bibliography or a directory by selecting, arranging, aggregating, and editing data, information, etc." but the editor in the sense of "Revising or clarifying the content, or selecting and putting together works or parts of work, by one or more creators is [specifically] excluded."  That, in addition to the fact that OpenAIRE/DataCite also specifically list Editor roles as those that need to listed as contributors, leaves me conclude that we should keep them as contributors only.  To get that done, we need to map all of the contributor roles to OpenAIRE.  I have completed that mapping and posted about it in a new Issue here: https://github.com/eprintsug/EPrintsOpenAIRE/issues/8  I will add the code for that later today.  We could always consider adding an option in the config, where editors would be treated differently, but by default, I think the plugin should follow cataloguing standards and OpenAIRE specifications.